src/main/java/org/broadinstitute/consent/http/models/Dataset.java

```diff
@@ -307,6 +322,22 @@ public boolean isStringMatch(@NonNull String query) {
     matchTerms.add(this.getDatasetIdentifier());
 
     if (Objects.nonNull(getProperties()) && !getProperties().isEmpty()) {
+      Optional<DatasetProperty> openAccessProp = getProperties()
+          .stream()
+          .filter((dp) -> Objects.nonNull(dp.getPropertyValue()))
+          .filter((dp) -> Objects.equals(dp.getPropertyName(), OPEN_ACCESS.toString()))
+          .findFirst();
+
+      if (openAccessProp.isEmpty()){
+        if (openAccess) {
+          return false;
+        }
+      }
+      else if (!Objects.equals(openAccessProp.get().getPropertyValue().toString(), Boolean.toString(openAccess)))
```

Comment:
Perhaps we can add a minor `TODO` to investigate whether we can switch to try to coerce `getPropertyValue` to a boolean instead.

Reply:
will do!